tools/vms-generator/vms-generator.go

```diff
@@ -53,7 +53,7 @@ func main() {
 
 	config, _, _ := testutils.NewFakeClusterConfigUsingKVConfig(&v1.KubeVirtConfiguration{
 		DeveloperConfiguration: &v1.DeveloperConfiguration{
-			FeatureGates: []string{"DataVolumes", "LiveMigration", "SRIOV", "GPU", "HostDisk", "Macvtap", "HostDevices"},
+			FeatureGates: []string{"DataVolumes", "LiveMigration", "SRIOV", "GPU", "HostDisk", "Macvtap", "HostDevices", "Sidecar"},
```

Comment:
I don't understand this change here. Should it have been there already and it was missing?

Reply:
We are doing:
https://github.com/kubevirt/kubevirt/blob/3a2b0b04ca06fbfa784c2ba3415f58e6b3ca8421/tools/vms-generator/vms-generator.go#L190-L200

but in the case of the VMI it was a real annotation that is completely missed by `ValidateVirtualMachineInstanceSpec` that is inspecting only the `spec` stanza.

In the case of a VM is now under `/spec/template/metadata/annotation` and so it's now correctly processed by `ValidateVirtualMachineSpec` and so we have to set the `Sidecar` FG if we want to get it admitted (even if just by the example generator).